### PR TITLE
Add new package: Odin

### LINKED ISSUE #8136: [Request] Add Odin language
### Information

**Name:** Odin Programming Language

**Description:** Odin is a general-purpose programming language with distinct typing, built for high performance, modern systems, and built-in data-oriented data types.

**Homepage:** https://github.com/odin-lang/Odin

**Download link(s):** https://github.com/odin-lang/Odin/releases

**Some indication of popularity/repute:** Stars: 2.6k | Watching: 68 | Forks: 203

I can make a PR for this if nobody else is interested in taking this 😄

### PR DESCRIPTION
**Pull request by tupini07**
This PR adds the package `Odin` to the bucket

Closes #8136

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
- Is there a GUI for this app? If not, shortcut is not needed.
- The prefix `dev-` can be omitted from the version string.
- I think the Main bucket is a better place for this.

File: bucket/odin.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "version": "dev-2022-03",
+    "description": "Odin is a general-purpose programming language with distinct typing, built for high performance, modern systems, and built-in data-oriented data types.",
+    "homepage": "https://odin-lang.org/",
+    "license": "BSD-3-Clause License",
```
Comment (issaclin32) [inline bucket/odin.json:5]:
```suggestion
    "license": "BSD-3-Clause",
```
See [SPDX identifiers](https://spdx.org/licenses/)

**Comment by tupini07** [comment]
Thanks for the feedback @rashil2000 !

> * Is there a GUI for this app? If not, shortcut is not needed.

No there isn't 😅 my bad. I've removed the _shortcut_ section

> * The prefix `dev-` can be omitted from the version string.

Done!

> * I think the Main bucket is a better place for this.

So, would you advise I close this PR and re-open it in the Main bucket? I'm a bit fuzzy as to what is the criteria for choosing whether a package goes in `main` or `extra`. I thought that since Odin doesn't have _release_ versions, just _dev_ versions, it was better to put it under `extra` 😄  but I defer to your expertise

**Comment by rashil2000** [comment]
> > * I think the Main bucket is a better place for this.
> 
> So, would you advise I close this PR and re-open it in the Main bucket? I'm a bit fuzzy as to what is the criteria for choosing whether a package goes in `main` or `extra`. I thought that since Odin doesn't have _release_ versions, just _dev_ versions, it was better to put it under `extra` 😄 but I defer to your expertise

I'm not sure what is the status of Odin project (with regards to readiness).

Actually, if a package does not have stable releases, it goes into the _Versions_ bucket. The _Main_ and _Extras_ are for stable, production-ready packages.

File: bucket/odin.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "version": "2022-03",
+    "description": "Odin is a general-purpose programming language with distinct typing, built for high performance, modern systems, and built-in data-oriented data types.",
```
Comment (rashil2000) [inline bucket/odin.json:3]:
```suggestion
    "description": "General-purpose programming language with distinct typing, built for high performance, modern systems, and built-in data-oriented data types.",
```

**Comment by tupini07** [comment]
oh got it! Thanks a lot. I'll close this PR and the relevant issue and open a new PR in the Versions bucket.

**Comment by rashil2000** [comment]
No worries!

**Pull request closed without merging**